src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -414,27 +414,58 @@
                                         (false? (:json_unfolding existing-field))))]
         (remove should-not-unfold? json-fields)))))
 
+(defn- sample-json-row-honey-sql
+  "Return a hoenysql used to get row sample to describe json columns.
+
+  If the talbe has a PK, try to fetch both first and last rows (see ##25744).
+  Else fetch the first n rows only"
+  [driver table-identifier json-field-names pk-identifier]
+  (binding [hx/*honey-sql-version* (sql.qp/honey-sql-version driver)]
+    (let [pk-expr          (sql.qp/maybe-wrap-unaliased-expr pk-identifier)
+          table-expr       (sql.qp/maybe-wrap-unaliased-expr table-identifier)
+          json-field-names (mapv sql.qp/maybe-wrap-unaliased-expr json-field-names)]
+      (if pk-identifier
+        {:select json-field-names
+         :from   [table-expr]
+         ;; mysql doesn't support limit in where, so we're using inner join here
+         :join  [[{:union [{:nest {:select   [pk-expr]
+                                   :from     [table-expr]
+                                   :order-by [[pk-expr :asc]]
+                                   :limit    (/ metadata-queries/nested-field-sample-limit 2)}}
+                           {:nest {:select   [pk-expr]
+                                   :from     [table-expr]
+                                   :order-by [[pk-expr :desc]]
+                                   :limit    (/ metadata-queries/nested-field-sample-limit 2)}}]}
+                  :result]
+                 [:=
+                  (sql.qp/maybe-wrap-unaliased-expr (hx/identifier :field :result :id))
+                  (hx/identifier :field table-identifier :id)]]}
```

Comment:
Can you explain why you use a join and not a simple union between two queries? Something like this seems simpler and should do the job?

```
SELECT <json-field-names>
FROM (
    SELECT *
    FROM table
    ORDER BY pk ASC
    LIMIT 250
) AS first_250
UNION ALL
SELECT <json-field-names>
FROM (
    SELECT *
    FROM table
    ORDER BY pk DESC
    LIMIT 250
) AS last_250
ORDER BY pk;
```

Reply:
as I noted on Line#430, mysql doesn't support limit in the subquery, so I switched to using join instead.

Reply:
oops. my bad, will update!

Reply:
updated in f82ebae506